improver_tests/spotdata/test_HeightAdjustment.py

```diff
@@ -0,0 +1,258 @@
+        grid_attributes=["vertical_displacement"],
+    )
+    return neighbour_cube
+
+
+@pytest.mark.parametrize("order", (True, False))
+@pytest.mark.parametrize(
+    "cube_name, expected",
+    (
+        (
+            "prob_cube",
+            [
+                [0.1, 0.7, 1],
+                [0, 0, 0.35555556],
+                [0.2, 0.3, 0.4],
+                [0.23888889, 0.27777778, 0.9],
+            ],
+        ),
+        (
+            "prob_cube_realizations",
+            [
+                [
+                    [0.1, 0.7, 1],
+                    [0, 0, 0.35555556],
+                    [0.2, 0.3, 0.4],
+                    [0.23888889, 0.27777778, 0.9],
+                ],
+                [
+                    [0.1, 0.7, 1],
+                    [0, 0, 0.35555556],
+                    [0.2, 0.3, 0.4],
+                    [0.23888889, 0.27777778, 0.9],
+                ],
+            ],
+        ),
+        (
+            "realization_cube",
+            [
+                [1000, 3671.91601, -200.0, 428.08399],
+                [3000.0, 9671.91601, 0.0, 528.08399],
+                [4000.0, 10671.91601, 30.0, 478.08399],
+            ],
+        ),
+        (
+            "percentile_cube",
+            [[1400, 3900, -500, 200], [2000, 9900, 0, 200], [3000, 10900, 30, 250]],
+        ),
+    ),
+)
+def test_cube(cube_name, expected, neighbour_cube, order, request):
+    """Tests plugin produces correct metadata and results for different input cubes"""
+    cube = request.getfixturevalue(cube_name)
+    coords_original = [c.name() for c in cube.dim_coords]
+    if order:
+        enforce_coordinate_ordering(
+            cube, [name, "realization", "percentile", "spot_index"]
+        )
+
+    coords_enforced = [c.name() for c in cube.dim_coords]
+    cube_units = cube.units
+    cube_title = cube.name()
+    try:
+        threshold_coord = cube.coord(name)
+    except CoordinateNotFoundError:
+        pass
+    result = SpotHeightAdjustment()(cube, neighbour_cube)
+    coords_result = [c.name() for c in result.dim_coords]
+
+    if order:
+        enforce_coordinate_ordering(result, coords_original)
+    assert coords_result == coords_enforced
+    assert result.units == cube_units
+    assert result.name() == cube_title
+    if cube_name == ("prob_cube" or "prob_cube_realizations"):
+        assert result.coord(name) == threshold_coord
+    np.testing.assert_allclose(result.data, expected)
+
+
+def test_prob_cube_threshold_unit(prob_cube, neighbour_cube):
+    """Tests that correct units and data are returned if probability cubes threshold is converted
+    to metres"""
+    prob_cube.coord(name).units = "km"
+    expected = [
+        [0.1, 0.7, 1.0],
+        [0, 0, 0.3999556],
+        [0.2, 0.3, 0.4],
+        [0.1002, 0.2000778, 0.9],
+    ]
+    cube_units = prob_cube.coord(name).units
+
+    result = SpotHeightAdjustment()(prob_cube, neighbour_cube)
+    assert result.coord(name).units == cube_units
```

Comment:
Just to be sure, you could include an additional assumption i.e.:
```
assert result.coord(name).units == "km"
```

Reply:
I've added this in as you suggest.